backend/dal/Services/Concrete/BuildingService.cs

```diff
@@ -187,7 +193,8 @@ public Building Update(Building building)
             // Only administrators can dispose a property.
             if (building.ClassificationId == 4 && !isAdmin) throw new NotAuthorizedException("Building classification cannot be changed to disposed.");
 
-            this.Context.Buildings.ThrowIfNotUnique(building);
+            // A building should have a unique name within the parcel it is located on.
+            existingBuilding.Parcels.ForEach(pb => this.Context.ThrowIfNotUnique(pb.Parcel, building));
```

Comment:
Shouldn't this be thrown if the new building names aren't unique, not the existing building names?

Reply:
I think that is what it is doing.